libs/entitlement/qa/common/src/main/java/org/elasticsearch/entitlement/qa/common/NetworkAccessCheckActions.java

```diff
@@ -59,4 +68,57 @@ static void socketConnect() throws IOException {
             socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
         }
     }
+
+    @SuppressForbidden(reason = "Testing entitlement check on forbidden action")
+    static void urlOpenConnectionWithProxy() throws URISyntaxException, IOException {
+        var url = new URI("http://localhost").toURL();
+        var urlConnection = url.openConnection(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(0)));
+        assert urlConnection != null;
```

Comment:
What are these asserts for?

Reply:
For the allowed path, checking that the connection was actually created and not nullified due to some (caught) exception (URL and URLClassLoader are weird this way, in some places)